tests/chainermn_tests/datasets_tests/test_scatter.py

```diff
@@ -39,13 +40,25 @@ def check_scatter_dataset(self, original_dataset, shuffle=False, root=0):
             # Test the content of scattered datasets
             joined_dataset = sum((sub_dataset[:]
                                   for sub_dataset in sub_datasets), [])
+
+            # NOTE: values in original_dataset must be int-like type
+            # (i.e. dtype('int64') or dtype('int32')
+            # This is necessary because numpy and cupy/chainerx have
+            # different behaviours on this.
+            # type(numpy.array([1])[0])
+            # =>  <class 'numpy.int64'>  # Scalar
+            # type(chainerx.array([1])[0])
+            # => <class 'chainerx.ndarray'>  # array of one element
+            joined_dataset = [int(e) for e in joined_dataset]
+            original_dataset = [int(e) for e in original_dataset]
             self.assertEqual(set(joined_dataset), set(original_dataset))
 
     def test_scatter_dataset(self):
         n = self.communicator.size
 
         for shuffle in [True, False]:
             for root in range(self.communicator.size):
+                self.communicator.mpi_comm.barrier()
```

Comment:
why do we need a barrier here?

Reply:
This line is unnecessary. forgot to remove debug code.

Reply:
fixed.